### PR TITLE
[SPARK-21947][SS] Check and report error when monotonically_increasing_id is used in streaming query

### PR DESCRIPTION
**Pull request by viirya**
## What changes were proposed in this pull request?

`monotonically_increasing_id` doesn't work in Structured Streaming. We should throw an exception if a streaming query uses it.

## How was this patch tested?

Added test.

### CODE REVIEW

**Comment by viirya** [comment]
retest this please.

**Comment by viirya** [comment]
cc @zsxwing for review. Thanks.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala [hunk 1]
```diff
@@ -516,6 +516,14 @@ class UnsupportedOperationsSuite extends SparkFunSuite {
   testOutputMode(Update, shouldSupportAggregation = true, shouldSupportNonAggregation = true)
   testOutputMode(Complete, shouldSupportAggregation = true, shouldSupportNonAggregation = false)
 
+  // Unsupported expressions in streaming plan
+  assertNotSupportedInStreamingPlan(
+    "MonotonicallyIncreasingID",
+    streamRelation.select($"a", MonotonicallyIncreasingID()),
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala]: nit. We can remove `$"a"` here.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by viirya** [comment]
ping @zsxwing Can you take a look? Thanks.

**Comment by zsxwing** [comment]
LGTM. Merging to master.

**Comment by viirya** [comment]
Thanks @zsxwing @dongjoon-hyun

**Pull request closed without merging**